python/pyarrow/filesystem.py

```diff
@@ -371,9 +371,99 @@ def walk(self, path, refresh=False):
                 yield tup
 
 
-def _sanitize_s3(path):
+class AzureDatalakeFileSystemWrapper(DaskFileSystem):
+    """
+    Wraps the Gen1 Azure Datalake Filesystem client.
```

Comment:
Can you add a short example of how this is supposed to be used? And mention the `dask-adl` package?